server/utils/tasks/listTable.ts

```diff
@@ -152,21 +152,32 @@ export type TabItem = {
   href: string
 }
 
-const tasksTabItems = (activeTab?: string, apArea?: string): Array<TabItem> => {
-  const hrefSuffix = apArea ? `&area=${apArea}` : ''
-  const allocatedHref = `${paths.tasks.index({})}?allocatedFilter=allocated${hrefSuffix}`
-  const unallocatedHref = `${paths.tasks.index({})}?allocatedFilter=unallocated${hrefSuffix}`
+const tasksTabItems = (hrefPrefix: string, activeTab = 'allocated'): Array<TabItem> => {
+  const [path, query] = hrefPrefix.split('?')
+  const urlParams = new URLSearchParams(query)
+
+  const allocatedParams = {
+    ...Object.fromEntries(urlParams),
+    allocatedFilter: 'allocated',
+  } as Record<string, string>
+  const { allocatedToUserId: _, ...unallocatedParams } = {
+    ...Object.fromEntries(urlParams),
+    allocatedFilter: 'unallocated',
+  } as Record<string, string>
```

Comment:
this assigns to a const named `unallocatedParams`? I'm guessing this is idiomatic TS but ...

Reply:
Yeah, it removes `allocatedToUserId` and assigns the rest to a const called `unallocatedParams`. It's not the prettiest, but it's the least faffy option I've found